packages/beacon-node/src/eth1/eth1DepositDataTracker.ts

```diff
@@ -112,7 +112,7 @@ export class Eth1DepositDataTracker {
   /**
    * Return eth1Data and deposits ready for block production for a given state
    */
-  async getEth1DataAndDeposits(state: BeaconStateAllForks): Promise<Eth1DataAndDeposits> {
+  async getEth1DataAndDeposits(state: CachedBeaconStateAllForks): Promise<Eth1DataAndDeposits> {
```

Comment:
can't see why you need cached here

Reply:
> can't see why you need cached here

`getEth1DepositCount()` needs cache to determine if the state is pre or post 6110 hence it requires `CachedBeaconStateAllForks` as its parameter. As such, all of its callers need to pass the cache around. This includes `getDeposits()` and `getEth1DataAndDeposits()`